crates/meroctl/src/common.rs

```diff
@@ -26,3 +29,28 @@ pub fn multiaddr_to_url(multiaddr: &Multiaddr, api_path: &str) -> EyreResult<Url
 
     Ok(url)
 }
+
+pub async fn get_response<T: Serialize>(
+    client: &Client,
+    url: Url,
+    request: Option<T>,
```

Comment:
this reads like "get the response of a `T`", but it's not, how about the `impl Trait` syntax?

```suggestion
pub async fn get_response(
    client: &Client,
    url: Url,
    request: Option<impl Serialize>,
```

Reply:
I tried it like that but had trouble because None is not `Seralize`, this change: `None::<()>` should fix that, thanks.